AwsCryptographicMaterialProviders/dafny/AwsCryptographyKeyStoreAdmin/src/InitializeMutation.dfy

```diff
@@ -73,6 +76,10 @@ module {:options "/functionSyntax:4" } InternalInitializeMutation {
       && input.Mutations.TerminalKmsArn.Some?
       ==>
         && KmsArn.ValidKmsArn?(input.Mutations.TerminalKmsArn.value)
+    ensures
+      && input.Mutations.TerminalHierarchyVersion.Some?
+      && input.Mutations.TerminalHierarchyVersion.value.v1?
+      ==> output.Failure?
```

Comment:
Food for thought -- should this be something we put in the spec, then link to the spec from here?
Not blocking

Reply:
This one of those feature development restrictions that I am not sure we want to permanently commit to.
There is no technical blocker from implementing an HV-2 to HV-1 logic;
but there is a logistical blocker (SDE Hours till Pen Test).

For now, I think we are committed to NOT supporting HV-2 to HV-1 for first delivery,
but MAY implement it later.

Based on that, I guess the Spec could say "the BKSA SHOULD allow for HV-2 to HV-1",
and then we fail the SHOULD.

However, @lucasmcdonald3 , you are spot on, I am hoping that our formal verification specifications can be turned into RFC style specifications.